macros/multiple_databases/dateadd.sql

```diff
@@ -0,0 +1,22 @@
+{%- macro dateadd(datepart, number, datetime_field) -%}
+{%- if target.type == 'snowflake' -%}
+    dateadd({{ datepart }}, {{ number }}, try_to_timestamp(to_varchar({{ datetime_field }})))
```

Comment:
If we know the datetime_field is a date or a datetime, we can use to_timestamp({{datetime_field}})